### DISCUSSION TITLE
GPS over bluetooth accuracy

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#232] [Q&A] [answered]
@ Emux. A peculiar observation with a gps bleutooth transfer between phones. 

For tests I mainly use an old A5 (orig A4.4) phone with a particularly bad quality inbuilt gps receiver.
So I use my standard phone in a pannier as gps source for the old phone by bluetooth gps data transfer.  
The bluetooth send app is "Share GPS" the receiving app "BleuGPS" (F-droid) or "Bleutooht gps" .
With the (bicycle) voice navigation both the phone turn announcements works fine. 
The announcements by the bluetooth old phone is just slightly delayed. (as expected)

The recalculate distance is set to 60m (10m to test) but does not seem to work correct. 
The recalculation always happens + 100m further away than the distance that is set.
_The precison report by the gps bluetooth connection probably so causes this trouble._

I alternatively tested by Locus map with the same 60m recalculation trigger distance.

The main phone by using the standard internal gps.
- The Locus recalculation trigger is **fairly precise** at 60m. 
   _I observe the Locus distance marker and notice a slightly variable precison reported._
- The Cruiser recalculation trigger is a little more variable and is after 60m.

The old phone tests using the bluetooth "mock location" transfer.

- The Locus announcement is **fairly precise** at 60m. 
   _I observe the Locus distance marker and notice a constant 100m precision report._
- The Cruiser recalculation trigger happens **WAY FURTHER** than +60m.
   _By set 60 m + 100m from the reported precision circle = 160m._
   _By set 10 m + 100m from the reported precision circle = 110m._ 

To record a track in Cruiser on the old phone (bluetooth gps) I have to set the minimum precision to OFF.

**Comment by devemux86** [comment]
@0709wiwiwi thanks for the interesting report.

Do you mean the Cruiser option "Minimal distance to recalculate route" in navigation rerouting settings?

Cruiser uses this selected distance + GPS accuracy (as reported by system) to detect if being off route.
So it is not precisely the selected off route distance, it could be larger depending on the GPS accuracy.

Does it help or do you mean something else?

**Reply by 0709wiwiwi** [reply]
Correct.  "**Minimal distance to recalculate route.**"    (By the way gps record results by the bleutooth connection are normal)
The solution for bleutooth gps transfers would be similar as the one you do offer in the recording conditions.   
No extra menu.  Remove **+ GPS accuracy** when in the record condition >Filters > **Minimum GPS accuracy is set to OFF**.

**Reply by devemux86** [reply]
The accuracy filter of GPS recording works directly with the GPS sensor (or GPS transfer in your example).

Off route detection (or route recalculation) is not such a low-level process.
And it is not connected with the GPS recording.

However, I am still not sure what benefit would we have if we use strictly the distance?

**Reply by 0709wiwiwi** [reply]
> However, I am still not sure what benefit we would have by using distance alone?

However, I am still not sure what benefit we do have by using "distance **+ GPS accuracy**".
_**GPS accuracy is too often quite worthless to be taken into account for a calculation.**_

Recalculation distance trigger  by distance only.
Simple rule and a solution for the bluetooth gps sources that do report 100m accuracy altough the source is more precise. 
I tested also by other bleutooth transfer apps and the reported accuracy is always inaccurate, not reliable and or unstable.  
 I must set the recording filter "Minimimum GPS accuracy is to OFF" as otherwise a Cruiser track record is impossible.

**Reply by 0709wiwiwi** [reply]
For your information. 
I tested with the external bleutooth app transfer with Locus **to compare**. 
In Cruiser you must use apps that offer the data by a "mock" gps location.  
- The **bleutooth gps** receiving app shows the sattelite constellation with a constant accuracy indication of **0m**.
- Locus app by this bleutooth "mock" app indicates a constant accuracy of **100m** ??

Locus has a built-in bleutooth gps receiver integrated in the app.
By the integrated module the accuracy indication is -- empty.

**Locus integrated bleutooth gps receiver result:** 
![image](https://github.com/devemux86/cruiser/assets/54167201/f3c45818-6d62-4e96-a7b2-0e5da78d14f2)

In the next image notice the 100m accuracy circle by bleutooth transfers apps with mock gps location.
These are than the sources that Cruiser has to work with........
**Mock gps app receiver result:**
![image](https://github.com/devemux86/cruiser/assets/54167201/27e7f1ce-ff59-4d41-a401-dc076983d9d9)

Anyway I have been doubting the reliability of the accuracy reports even by the usual smartphone gps setup for much longer.
In next image, 1.7m is reported horizontally (12m accuracy on map circle) and the height report (always delicate by GPS) deviates + 40 m.  This while the reported horizontal on map gps position deviates +/- 100m away from the real position.
**Bad quality gps receiver phone result:**
![image](https://github.com/devemux86/cruiser/assets/54167201/797abb7b-efb1-4d2a-aff4-e07d4f27fd5e)

My experiences so far ?  Those shown accuracy reports are rather toy quality and better not to be taken too seriously and of little use.

References:  
https://forum.locusmap.eu/index.php?topic=8264.msg70042#msg70042
https://forum.locusmap.eu/index.php?topic=1217.msg7251#msg7251
https://wiki.openstreetmap.org/wiki/Test_the_accuracy_of_GPS_devices

"_The producers of consumer-devices mostly don't give any information on the used hard- and software. Details on signal correction ("WAAS" etc.) are often indefinite to misleading. Also, the displayed "accuracy" of some devices is not defined and therefore **rather useless**._"   **Rather useless.....so true ;-)**

**Comment by devemux86** [comment]
@0709wiwiwi thanks for the extra details.

I suppose we can skip the GPS accuracy and its non accurate reports in the "off route detection" and see how it goes.

**Reply by devemux86** [reply, marked as answer]
> I suppose we can skip the GPS accuracy and its non accurate reports in the "off route detection" and see how it goes.

Available in [Cruiser 3.0.25](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

You can adjust the off route distance in "Settings | Navigation | Recalculate route | Minimal distance to recalculate route".

**Reply by 0709wiwiwi** [reply]
Tested v 3.0.25 gps over bleutooth gps share.  The autorecalculation trigger distance now is as set in Cruiser.  Thanks !
